driver-core/src/test/unit/com/mongodb/client/model/AggregatesSpecification.groovy

```diff
@@ -847,6 +849,58 @@ class AggregatesSpecification extends Specification {
         }''')
     }
 
+    def 'should render $vectorSearch'() {
```

Comment:
Could we write these new tests in Java? We have been doing this (see `AggregatesTest.java`) for new test cases, and sometimes even when making changes to existing ones.

Reply:
`AggregatesTest` contains integration tests, it appears to be a successor of `AggregatesFunctionalSpecification`. `AggregatesSpecification.groovy` contains unit tests, and this PR added unit tests to this file.